src/iqm/qiskit_iqm/iqm_provider.py

```diff
@@ -82,6 +83,30 @@ def max_circuits(self, value: Optional[int]) -> None:
         self._max_circuits = value
 
     def run(self, run_input: Union[QuantumCircuit, list[QuantumCircuit]], **options) -> IQMJob:
+        (
+            circuits_serialized,
+            qubit_mapping,
+            calibration_set_id,
+            shots,
+            max_circuit_duration_over_t2,
+            heralding_mode,
+            circuit_metadata,
+        ) = self._prepare_circuits(run_input, **options)
+        # NOTE: make sure client.submit_circuits input is exactly the same as client.create_run_request input in
+        # create_run_request
+        job_id = self.client.submit_circuits(
+            circuits_serialized,
+            qubit_mapping=qubit_mapping,
+            calibration_set_id=calibration_set_id if calibration_set_id else None,
+            shots=shots,
+            max_circuit_duration_over_t2=max_circuit_duration_over_t2,
+            heralding_mode=heralding_mode,
+        )
+        job = IQMJob(self, str(job_id), shots=shots)
+        job.circuit_metadata = circuit_metadata
+        return job
```

Comment:
Perhaps the flow should be
```
def IQMBackend.run:
    self.create_run_request
    self.client.submit_run_request
    return job

def IQMBackend.create_run_request -> RunRequest:
    (maybe transpile)
    self._prepare_circuits (inlined)
    return self.client.create_run_request
```
so we would not use `client.submit_circuits` at all? I think this would simplify `IQMBackend.run` a bit.

Reply:
And `IQMBackend.create_run_request` would always be on the execution pathway.

Reply:
Same in Cirq I guess.

Reply:
Good idea, changed now.